Windows: Use common path for standard library

bryphe opened the pull request:
First off, thank you for all the incredible work you've done with the OCaml compiler!

I've been experimenting with the OCaml compiler as part of the https://github.com/esy/esy project, in particular, facilitating a smooth Windows experience. 

I noticed that the standard library path was different on Windows vs OSX / Linux:
- Windows: `<ocaml>/lib`
- OSX / Linux: `<ocaml>/lib/ocaml`

It seems like it would make sense for these to be consistent cross-platform. Looking at the history of the `Makefiles`, it looks like the `LIBDIR` was set a while ago - so I'm curious if perhaps it was changed in OSX / Linux, but not on Windows. For most tools and applications, it may not be important, as the `OCAMLLIB` variable can be used - but `esy` is a somewhat special case as it implements a sandbox isolation model, and needs strict control over the environment. Having a common path from the `ocaml` root means less special casing per-platform.

In addition, I wonder if there are any compatibility issues making this change. Please let me know if you'd like any information or have any feedback on this change. Thank you!

bryphe committed:
Use '<PREFIX>/lib/ocaml' for standard library instead of '<PREFIX>/lib' for cross-platform consistency

nojb commented:
@shindere

nojb commented:
In principle this seems a harmless change: Windows users typically modify their Makefiles anyway before building. But let's see what the build system experts think about it.

shindere commented:
Well, not being a user of Windows I have not strong opinion on this.
Perhaps @dra27 wants to say something?

dra27 commented:
In opam, this is already done (see [here](https://github.com/ocaml/opam/blob/master/shell/bootstrap-ocaml.sh#L114)) because it's such a nuisance having the layout being different.

I can't see how this would cause a problem - please add a `Changes` entry, though.

bryphe commented:
Wonderful! Thanks all for the feedback, and thank you @dra27 for the review.

I've added a `Changes` entry (I added your name too @jordwalke since your suggestion prompted this change). Please let me know if you'd like any additional revisions.

bryphe committed:
Add Changes entry

nojb commented:
Please squash "Resolve merge conflicts" onto your first commit and we can merge this.

bryphe commented:
Done! Thanks @nojb .
